### PR TITLE
HARVESTER: Add rke1 cloud provider for harvester

### PR DESCRIPTION
**Pull request by n313893254**
Proposed changes
======

image

Types of changes
======

- New feature (non-breaking change which adds functionality)

Linked Issues
======

- https://github.com/rancher/rancher/issues/35000
- https://github.com/harvester/harvester/issues/1396

 

Further comments
======

Waiting these backend PR to merge: 
- https://github.com/rancher/rke/pull/2733
- https://github.com/rancher/charts/pull/1595

### CODE REVIEW

**Comment by n313893254** [comment]
[Backend PR](https://github.com/rancher/rke/pull/2733) has merged.

@richard-cox Could you help review this PR? thanks. cc @nwmac

**Review by richard-cox — CHANGES_REQUESTED** [review]

File: lib/shared/addon/components/cru-cloud-provider/component.js [hunk 1]
```diff
@@ -230,6 +230,11 @@ export default Component.extend({
     let cluster               = get(this, 'cluster');
     let config                = get(cluster, 'rancherKubernetesEngineConfig') || set(cluster, 'rancherKubernetesEngineConfig', {});
 
+    const cloudCredentialId = get(this, 'model.nodeTemplates.firstObject.cloudCredentialId')
```
[the remark this answers is not in the document]
Comment (n313893254) [reply lib/shared/addon/components/cru-cloud-provider/component.js]: Done

**Comment by n313893254** [comment]
Hi, @richard-cox. Thanks for your review, I fixed the comments, PTAL, thanks.

**Review by richard-cox — CHANGES_REQUESTED** [review]
TBH I'm not sure of the flow here, however when selecting `Harvester (Out-of-tree)` in an install without any harvester instances it flashes up a notice on the right which is quickly removed and doesn't change the radio button selection.

The underlying error is 
```
Assertion Failed: Cannot call get with 'harvestercredentialConfig.clusterId' on an undefined object.
```
So i'm guessing there's no
```
const cloudCredential = Ember.get(this, 'globalStore').getById('cloudCredential', cloudCredentialId);
```

Should the option be visible / enabled if it's not possible to use? The creds are something supplied with the node template above correct, if so maybe the option can be disabled until set?

**Comment by n313893254** [comment]
Thanks a lot, the comment truely helps me. Now the option can not be selected before select a node template first. And added a tooptip to notify user should select a node template.

image

**Review by richard-cox — APPROVED** [review]

**Pull request merged**